pkg/trace/api/api.go

```diff
@@ -121,9 +122,37 @@ func (r *HTTPReceiver) buildMux() *http.ServeMux {
 	mux.HandleFunc("/v0.5/stats", r.handleStats)
 	mux.Handle("/profiling/v1/input", r.profileProxyHandler())
 
+	r.attachDiscoveryHandler(mux)
 	return mux
 }
 
+func (r *HTTPReceiver) attachDiscoveryHandler(mux *http.ServeMux) {
+	refPatterns := reflect.ValueOf(mux).Elem().FieldByName("m").MapKeys()
```

Comment:
Other solution that is probably cleaner is to wrap the mux handler in a custom struct.